utopia-remix/app/models/project.server.ts

```diff
@@ -23,12 +23,15 @@ export async function listProjects(params: { ownerId: string }): Promise<Project
   })
 }
 
-export async function getProjectOwnerById(params: { id: string }): Promise<string | null> {
+export async function getProjectOwnerById(
+  params: { id: string },
+  config: { allowDeleted: boolean } = { allowDeleted: false },
+): Promise<string | null> {
   const project = await prisma.project.findFirst({
     select: { owner_id: true },
     where: {
       proj_id: params.id,
-      OR: [{ deleted: null }, { deleted: false }],
+      OR: config.allowDeleted ? [] : [{ deleted: null }, { deleted: false }],
```

Comment:
this looks fine, but _**please**_ add a test for this. Prisma can be surprisingly non-intuitive and we 10000000% want to be sure we don't get incorrect selections on soft-deleted projects

Reply:
case in point for super scary, counter-intuitive Prisma behaviors: https://github.com/concrete-utopia/utopia/blob/77adcec58e49177f58b5c8a514530bb5b38353bf/utopia-remix/app/db.server.ts#L18-L24

Reply:
yes! added tests 🙏